src/api/main/LoginListener.ts

```diff
@@ -14,17 +19,32 @@ export interface ILoginListener {
 	 */
 	onFullLoginSuccess(): Promise<void>
 
-	onLoginError(): Promise<void>
+	/**
+	 * call when the login fails for invalid session or other reasons
+	 */
+	onLoginFailure(reason: LoginFailReason): Promise<void>
+
+	/*
+	* call when retrying full login
+	 */
+	onRetryLogin(): void
 
 	/**
 	 * Shows a dialog with possibility to use second factor and with a message that the login can be approved from another client.
 	 */
 	onSecondFactorChallenge(sessionId: IdTuple, challenges: ReadonlyArray<Challenge>, mailAddress: string | null): Promise<void>
+
+	/**
+	 * true if the last full login attempt failed
+	 * may revert to false when retrying.
+	 */
+	getFullLoginFailed(): boolean
 }
 
 export class LoginListener implements ILoginListener {
 
 	private loginPromise: DeferredObject<void> = defer()
+	private fullLoginFailed: boolean = false
```

Comment:
If we are okay with adding state here we might as well track it more explicitly:
 - Not logged in
 - Partially logged in (attempting full login)
 - Partially logged in (full login failed)
 - Fully logged in

And then we can consume it more easily together with WS state in offline indicator VM

Reply:
Generally on the client side the holder of login state is `LoginController`, we didn't make it listener because it would introduce circular dependency between `LoginFacade` and `LoginController`.

I'm starting to think whether we should do the same thing as we did on the worker side (`UserFacade` as a holder of state) with client side.

But when :crying_cat_face:

Reply:
> But when crying_cat_face
we're doing this later.